### DISCUSSION TITLE
Inverse link function

### DISCUSSION THREAD

**Discussion by raffaele-morelli** [gavinsimpson/gratia#194] [Q&A] [answered]
I have a gam model with this family
```
Family: gaussian 
Link function: log 
```
Is there any chance for appraise to plot linear predictor on x axis and taking into account the inverse link function? I saw an issue about the use of trans parameter but can't see how :-)

Any suggestions?
Regards

@gavinsimpson

**Comment by gavinsimpson** [comment, marked as answer]
Why would you want to do that within `appraise()`? The `draw()` methods for GAMs have arguments `constant` and `trans` that would allow you to plot each individual smooth as if it were on the response scale, in your case you'd need to pass `constant = coef(m)[1], trans = inv_link(m)` to `draw(m, ....)` and even then this is only really OK with a single smooth...

As `appraise()` is for diagnosing potential problems with the model, and one of the tools we have for that is to plot the linear predictor on the link scale, there isn't an option to plot on the response scale --- it would be extraordinarily difficult indeed to discern if the mean variance relationship say was the correct one. Also, we'd have to define residuals on the actual response scale, which are rarely useful from a model diagnostic view point generally for models beyond Gaussian ones.

If you want to plot the residuals vs the linear predictor on the response scale, you can do:

```r
library("mgcv")
library("gratia")
library("ggplot2")

dat <- data_sim("eg1", n = 400, dist = "normal", scale = 2, seed = 2)
mod <- gam(y ~ s(x0) + s(x1) + s(x2) + s(x3), data = dat,
  family = gaussian(link = "log")) # <-- wrong fmaily!

fv <- fitted_values(mod, scale = "response") |>
  add_residuals(mod, type = "response")

fv |>
  ggplot(aes(x = fitted, y = .residual)) +
  geom_point() +
  geom_smooth(method = "loess", col = "red")
```

Or was there something more specific from `appraise()` that you are trying to do?

**Reply by raffaele-morelli** [reply]
I am using `appraise() ` for model diagnostic and yes, I've finally managed to split result into single plots as you suggested.
Nothing more specific. 

```
# don't want this
residuals_linpred_plot(mod)

# I was asked to plot linear predictor against residuals on the response scale like this
resid <- residuals(mod, type = "deviance")
lp <- fitted(mod, scale = "response")
dfS <- data.frame(linpred = fv, resid = resid)

ggplot(data = dfS, aes(x = linpred, y = resid)) + 
  geom_point() + 
  geom_smooth(method = "loess", col = "red", se = FALSE) +
  labs(x = "Linear predictor", y = "Residuals")
```

You wrote `family = gaussian(link = "log")) # <-- wrong family!`, may I ask why?

**Reply by gavinsimpson** [reply]
You could just as easily use response residuals here as this is a Gaussian model.

> You wrote family = gaussian(link = "log")) # <-- wrong family!, may I ask why?

because the data were simulated assuming `gaussian(link = "identity")`, so the model fitted is wrong for this simulation of data
